src/parser/warlock/demonology/modules/pets/normalizers/SummonOrderNormalizer.js

```diff
@@ -0,0 +1,68 @@
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+import SPELLS from 'common/SPELLS';
+
+const MS_BUFFER = 100;
+
+class SummonOrderNormalizer extends EventsNormalizer {
+
+  normalize(events) {
+    const _events = [];
+
+    events.forEach((event, idx) => {
+      _events.push(event);
+
```

Comment:
Seeing as both conditionals do the same (check for some condition, then iterate backwards from the event up to MS_BUFFER ms, then if some other condition is true, swap the events), you might be able to combine them together. I think this could work:

```javascript
// make a new helper
if (this.isDogSummon(event) || this.isTyrantSummon(event)) {
   const castTimestamp = event.timestamp;

   for (let previousEventIndex = idx; previousEventIndex >= 0; previousEventIndex -= 1) {
      const previousEvent = _events[previousEventIndex];
      if ((castTimestamp - previousEvent.timestamp) > MS_BUFFER) {
         break;
      }
      if (this.isSharpenedDreadfangs(previousEvent, event) || 
          this.isDemonicConsumption(previousEvent, event) || 
          this.isDemonFire(previousEvent, event)) {
        this.swapEvents(_events, previousEventIndex, previousEvent);
        break;
      }
   }
}
```

Now whether that is a *good idea* to join it together, is a different question. I think it's fine to leave it like you already have, I think it might be more readable.

Reply:
I agree it is more readable this way, I found a sample code in resto normalizer, maybe there should be a more general solution which could be reused?

anyway, updated the PR

Reply:
About other pets, I did not see an issue with other pets so far, to limit the scope of this PR, I would keep it this way, until there is a bug with other pets as well.